packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1404,6 +1404,28 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
     return child;
   }
 
+  /// Transition that matches with the next route. Will be null if there is no
+  /// next route or a next route is not provided. If a non-null value exists,
+  /// then it will be used in place of [buildTransitions].
+  ///
+  /// TODO: clean this up if viable
+  /// TODO: put in explaination how to never allow this behavior?
+  DelegatedTransitionBuilder? get nextRouteTransition => null;
+
+  /// docs
+  Widget buildFlexTransitions(
```

Comment:
Discussed in person, @MitchellGoodwin was still looking into if this was the best way to do this.

Reply:
I think this is good now.